### DISCUSSION TITLE
'Draw' behaviour changes in Version 7

### DISCUSSION THREAD

**Discussion by whugemann** [ImageMagick/ImageMagick#4893] [Help]
I have some code to transfer from IM version 6 to version 7 and which now leads to diffferent results. The problem can be illustrated by the following sample code:
`convert -size 100x100 xc:darkgrey  -draw "circle 50,50 0,50" -negate -fill none -stroke red -draw "circle 50,50 50,75" im6.jpg
magick -size 100x100 xc:darkgrey  -draw "circle 50,50 0,50" -negate -fill none -stroke red -draw "circle 50,50 50,75" im7.jpg`

Whereas IM 6 draws the red circle on the white area, IM 7 sourrounds it by a black rectangle. 

I guess that this has something to do with transparency, by I cannot find the right operation to yield the same result as in IM 6.

**Comment by snibgo** [comment]
See the documentation on upgrading to v7 at http://www.imagemagick.org/script/porting.php

I suggest you insert "-channel RGB" before drawing, and "+channel" after drawing.

**Reply by whugemann** [reply]
Thanks for pointing me the right direction. Inserting "-channel RGB" solved the problem. 
"+channel" (to reverse the channel settings, I guess) is not needed over here, as the target format it JPEG.
I remembered vaguely that in IM 7 there is an alpha channel by default, but I had no idea what to do about it ...
I put your link to my links collection!

**Comment by snibgo** [comment]
Yes, in this case, "+channel" to restore the default is not needed. Whenever you have "-channel", it is good practise to have a matching "+channel".
